---
title: "TIST"
subtitle: "- Transcriptome and Histopathological Image Integrative Analysis for Spatial Transcriptomics "
author: 
- name: "Yiran Shan"
  affiliation: &1 MOE Key Laboratory of Bioinformatics, BNRIST Bioinformatics Division, Department of Automation, Tsinghua University, Beijing 100084, China
- name: "Qian Zhang"
  affiliation: *1
- name: "Wenbo Guo"
  affiliation: *1
- name: "Yanhong Wu"
  affiliation: *1
- name: "Hongyi Xin"
  affiliation: &2 UM-SJTU Joint Institute, Shanghai Jiao Tong University, Shanghai, 200240, China &3Department of Automation, Shanghai Jiao Tong University, Shanghai 200240, China 
- name: "Qiuyu Lian"
  affiliation: &2 UM-SJTU Joint Institute, Shanghai Jiao Tong University, Shanghai, 200240, China &3Department of Automation, Shanghai Jiao Tong University, Shanghai 200240, China 
- name: "Jin Gu"
  affiliation: *1
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{my-vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(TIST)
```


<!-- ```{r setup} -->
<!-- library(TIST) -->
<!-- ``` -->


```{r style, echo = FALSE, results = 'asis'}
  BiocStyle::markdown()
```

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```



# Introduction

Sequencing-based spatial transcriptomics (ST) is an emerging technique to study in situ gene expression patterns at whole-genome scale.  However, this technique still has several limitations, especially high data dropouts and local molecular diffusion effects. 
Except the transcriptomic data, the technique usually generates matched histopathological images for the same tissue sample.  The image data are of high spatial continuity and resolution, which can provide complementary cellular phenotypical information with the noisy ST data.

Here, we propose a novel ST data analysis method called TIST (Transcriptome and histopathological Image integrative analysis for Spatial Transcriptomics) by integrating the information from the sequencing-based ST data and the histopathological images. TIST uses Markov random field to learn the macroscopic cellular features from histopathological images, and devises a random-walk-based strategy to integrate the extracted image features, the transcriptomic features and the location information for spatial cluster (SC) identification and gene expression enhancement. 
The workflow of  `TIST` 展示如下：
![TIST](https://user-images.githubusercontent.com/20694388/158058718-dbccdbe4-ad86-4214-8650-5c4c0792cb60.png)


# Basic instructions

## System Requirements
* R version: >= 3.5.0

## Installation
Firstly, please install or update the package `devtools` by running 
```{r eval=FALSE}
install.packages("devtools")
```

Then the `TIST` can be installed via
```{r eval=FALSE}
library(devtools)
devtools::install_github("ShanYiran/TIST")
```
<span style="color:red">**Hint**</span>: 

1) A dependent package `NNLM` was removed from the CRAN repository recently, so an error about it may be reported during the installation. 
If so, you can install a formerly available version manually from its [archive](https://cran.r-project.org/src/contrib/Archive/NNLM/).



## Loading
```{r eval=FALSE}
library(TIST)
```

## Data preparation
The `TIST` is mainly designed for 10X Genomics platform, 
and it requires a data folder containing the results generated by the software 
[`Space Ranger`](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/what-is-space-ranger).
In general, the data folder needs to be organized as following which is the output of `Space Ranger`:

```
    /sampleFolder
    ├── filtered_feature_bc_matrix
    │   ├── barcodes.tsv.gz
    │   ├── features.tsv.gz
    │   └── matrix.mtx.gz
    ├── spatial
    │   ├── tissue_hires_image.png
    │   ├── tissue_lowres_image.png
    │   ├── scalefactors_json.json
	│	└── tissue_positions_list.csv
    └── web_summary.html
```
For other droplet-based platforms, the data folder should be prepared likewise.

## Quick start

Here, we provide an [example data](V1_Adult_Mouse_Brain) of 
Mouse brain from 10X Genomics. Users can download it and run following scripts 
to understand the workflow of `TIST`.

```{r eval=FALSE}
library(TIST)

#The address where the Mask file is stored. You can use no mask to generate an all 1 matrix. The more recommended approach is generate this mask file use our python code in this package.
Maskfile <- "./data/V1_Adult_Mouse_Brain/spatial/Imginit/mask1.txt"
#The address where the Image file is stored. 
imagefile <- "./data/V1_Adult_Mouse_Brain/spatial/tissue_hires_image.png"
#The address where the results stored.
savePath <- "./data/V1_Adult_Mouse_Brain/test_package_results/"
#The address where the local file of all spots is stored.
spaceFile <- "./data/V1_Adult_Mouse_Brain/spatial/tissue_positions_list.csv"
#The address where the RNA-seq file is stored.
exprPath = "./data/V1_Adult_Mouse_Brain/filtered_feature_bc_matrix/"
#The scale to map the spots coordinate in spaceFile with image.
sacle_score <- 0.17011142

#run Meta_St_img_unsupervised
Spot_manifest_imgunsup <- Meta_St_img_unsupervised(Maskfile = Maskfile,
                                                     imagefile = imagefile,
                                                     spaceFile = spaceFile,
                                                     exprPath = exprPath,
                                                     colors = NULL,
                                                     savePath = savePath,
                                                     Method = "walktrap",
                                                     sacle_score = sacle_score)
```

Running the `Meta_St_img_unsupervised` script will generate some files/folders as below:

1. **Spot_manifest_imgunsup.csv** : 每个样本的分类标签信息.
2. **ST_imgunsupnet.RDS** : 用于下游分析的多模态网络.
3. **McRFlabel.RDS** : 图像初始化标签结果.
4. **expr_obj.RDS** : 表达信息的seurat类型数据
5. **ST_meta_imgunsup_plot.png** ： 聚类可视化图

方法接下来整合一个cluster内的所有spot，得到整合后的meta 表达矩阵，你可以执行 Meta_expr_matrix

```{r eval=FALSE}
Mc_manifest <- Meta_expr_matrix(exprPath = exprPath,Spot_manifest = Spot_manifest_imgunsup,
                                  imagefile = imagefile,
                                  savePath = savePath,merge_method = "mean")
```			
Running the `Meta_expr_matrix` script will generate some files as below:

1. **Mc_obj.RDS** : 整合后的meta 表达seurat数据.
2. **Mc_manifest** : 用于下游分析的多模态网络.
3. **MC_idents.RDS** : 图像初始化标签结果.
4. **expr_obj.RDS** : 表达信息的seurat类型数据

接下来你可以进行下游分析获取空间差异基因，我们提供了和SPARK方法的对比方法

```{r eval=FALSE}
Spark_methods(exprPath,
			  spaceFile,
			  savePath)
#基因表达信息的数据路径
SC_obj_file <- paste0(savePath,"expr_obj.RDS")
#聚类表达信息数据路径，用于展示基因表达类内相似性
MC_obj_file <- paste0(savePath,"Mc_obj.RDS")
#多模态网络数据
netfile <- paste0(savePath,"ST_imgunsupnet.RDS")
SPARK_file <- paste0(savePath,"SPARK.rds")
mc_marker_plot(gl = NULL,MC_obj_file,Mc_manifest,Spot_manifest_imgunsup,savePath)
#表达信息的seurat类型数据
SC_obj <- readRDS(SC_obj_file)
SpaceDiffGene(SC_obj = SC_obj,
                Spot_manifest= Spot_manifest_imgunsup,
                savePath = savePath,
                MC_obj_file = MC_obj_file,
                Mc_manifest = Mc_manifest,
                methods= "walktrap",
                SPARK_file = SPARK_file,
                netfile = netfile)

}

```
Running the `SpaceDiffGene` script will generate some files as below:

1. **TIST_markers.csv** : 差异表达基因结果文件.
2. **TIST_markers.RDS** : 差异表达基因结果文件.




